test/integrations/component.test.ts

```diff
@@ -79,26 +80,32 @@ if (!opts.generate || opts.generate === 'false') {
       const { url, method, data: reqData, ...opts } = axiosMock.httpReq;
       const { data, headers, status } = axiosMock.httpRes;
 
+      const headersAsymMatch = {
+        asymmetricMatch: function (actual) {
+          return isMatch(actual, opts.headers);
+        },
+      };
+
       switch (method.toLowerCase()) {
         case 'get':
           // @ts-ignore
-          mock.onGet(url, reqData, opts.headers).reply(status, data, headers);
+          mock.onGet(url, reqData, headersAsymMatch).reply(status, data, headers);
```

Comment:
do we need to same with reqData like headers use headersAsymMatch?
also for url should we use "new RegExp(`(${url}`)" this way it will match contains also.

Reply:
Don't think that is required. I think the data match here should be exact as there is no additional transformation being done.

The same logic for URL.

Reply:
If there are any cases that we see where in something else is happening inside Axios, we can probably do an `asymmetricMatch`

Reply:
idea here is, we don't want to mention all the details in test cases and only mention required things so that it improves readability

Reply:
The point is, the data or url might involve transformational changes where in exactness is more necessary. Headers might contain  content-type or authorization which doesn't involve a lot of complexity. Hence headers having asymmetric match makes sense but having data or url, I'm not 100% convinced about it. But do let me know if you think otherwise

Reply:
we can defer this and add it later if use case arises too